### PR TITLE
CHE-158 - Consolidate a few external PRs

### PR DESCRIPTION
**Pull request by gm-cht**
Documentation links:

    JIRA Ticket(s): [CHE-158](https://cloudhealthtech.atlassian.net/browse/CHE-158)

Proposed changes:

- Add Release.namespace in service account -- Will make sure Pods will run on the same namespace as entered in helm chart install
- Add pod annotations support in deployment
- Remove 'watch' from ClusterRole permissions. Only get and list are needed.
- Updated version to 1.0.1 and added contributors list. All changes to Chart.yaml. It shows up in index.yaml.

Test Plan and Risk Assessment:

How did you test this locally?

 - Use minikube and local collection endpoint. Use the new helm chart to install collector on local minikube and check collection endpoint logs
 - Copy the helm chart (.tgz file of 1.0.1) to install collector on a kubernetes cluster running on Azure Kubernetes Service. Make sure cluster shows up as healthy. Screenshots below.
 
image
image

How will you test this in production?

    Will test helm chart on GCP Kubernetes Service.

Rollback Plan:

    Remove 1.0.1 helm chart (.tgz file) from our repo. This will ensure we will install the previous version.

### CODE REVIEW

File: cloudhealth-collector/templates/clusterrole.yaml [hunk 1]
```diff
@@ -16,4 +16,4 @@ rules:
     - "nodes"
     - "services"
     - "namespaces"
-  verbs: ["get", "watch", "list"]
+  verbs: ["get", "list"]
```
Comment (gm-cht) [inline cloudhealth-collector/templates/clusterrole.yaml:19]: "watch" is higher level of permissions we do no use or need in the future.

Comment (gm-cht) [inline cloudhealth-collector/templates/clusterrole.yaml:19]: "watch" is higher level of permissions we do no use or need in the future.

File: cloudhealth-collector/templates/clusterrolebinding.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ metadata:
 subjects:
 - kind: ServiceAccount
   name: {{ include "cloudhealth-collector.fullname" . }}
-  namespace: default
+  namespace: {{ .Release.Namespace }}
```
Comment (gm-cht) [inline cloudhealth-collector/templates/clusterrolebinding.yaml:16]: Mandatory field and thus had to change from 'default' to Release.namespace. This takes the namespace entered during helm install using `-n` option

Comment (gm-cht) [inline cloudhealth-collector/templates/clusterrolebinding.yaml:16]: Mandatory field and thus had to change from 'default' to Release.namespace. This takes the namespace entered during helm install using `-n` option

File: cloudhealth-collector/templates/deployment.yaml [hunk 3]
```diff
@@ -18,6 +18,10 @@ spec:
       {{- include "cloudhealth-collector.selectorLabels" . | nindent 6 }}
   template:
     metadata:
+      {{- with .Values.podAnnotations }}
```
Comment (gm-cht) [inline cloudhealth-collector/templates/deployment.yaml:21]: Pod Annotations are present in values.yaml but were never used till this change is made

Comment (gm-cht) [inline cloudhealth-collector/templates/deployment.yaml:21]: Pod Annotations are present in values.yaml but were never used till this change is made

**Review by wxia20 — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -32,6 +32,13 @@ $ helm repo add cloudhealth https://cloudhealth.github.io/helm/
 $ helm install cloudhealth-collector --set apiToken=<Unique Customer API Token>,clusterName=<Cluster Name> cloudhealth/cloudhealth-collector
 ```
 
+To install the chart with the release name `cloudhealth-collector` in a particular namespace `<test_namespace>`, run the following command:
```
Comment (gm-cht) [inline README.md:35]: New instructions for installing in a namespace that is not `default`

File: index.yaml [hunk 5]
```diff
@@ -3,12 +3,36 @@ entries:
   cloudhealth-collector:
   - apiVersion: v2
     appVersion: 2.0.175
-    created: "2022-01-10T11:09:38.340238-05:00"
+    created: "2022-03-08T12:19:36.194034-05:00"
+    description: A Helm chart for Kubernetes
+    digest: 94a699b6c6dc40f74f8ece8aae470afb83882e7b00d40da9862593272d2e31e5
+    home: https://cloudhealth.vmware.com/
+    maintainers:
+    - email: gmudumbai@vmware.com
```
Comment (bswetalina) [inline index.yaml:11]: Do we want to keep individual emails here or instead an alias?

File: README.md (hunk 4 above)
Comment (wxia20) [reply README.md:35]: Would it be better to use `<target_namespace>` or `<namespace>` instead of `<test_namespace>`?

**Review by wxia20 — COMMENTED** [review]
Would it be better to use `<target_namespace>` or `<namespace>` instead of `<test_namespace>`?

File: README.md (hunk 4 above)
Comment (wxia20) [reply README.md:35]: Would it be better to use `<target_namespace>` or `<namespace>` instead of `<test_namespace>`?

File: index.yaml (hunk 5 above)
Comment (gm-cht) [reply index.yaml:11]: yes. makes sense.. I was debating myself and thought the team deserves mention. I guess giving personal email is not good idea..

**Review by trjaskoviak — APPROVED** [review]

**Review by wxia20 — APPROVED** [review]

**Review by kscherme — APPROVED** [review]

**Review by bswetalina — APPROVED** [review]

**Pull request merged**